### ISSUE TITLE
Fix Bugs in Add Bond Dropdowns and Update Behavior

### ISSUE THREAD

**Issue by TharunKumarrA** [TharunKumarrA/Molecule-Visualiser-DSA#9]
## Description

There are two issues with the **Add Bond** feature that need to be resolved:
1. The dropdown menus include unnecessary placeholders like `A)` to `A5`, which are not valid selections.
2. You need to click the **Add Bond** button twice for changes to reflect in the 3D plot.

---

## Acceptance Criteria

- Remove unnecessary placeholders (`A)`, `A1`, `A2`, etc.) from the **Add Bond** dropdown menus.
- Ensure that changes to bonds are reflected in the 3D plot after a single click of the **Add Bond** button.
- Verify the functionality works seamlessly for all valid bond selections.

---

### Dropdown Cleanup
1. Locate the logic that populates the **Add Bond** dropdown in `/src/components/EditMolecule.jsx` (or the relevant file).
2. Remove or filter out the unnecessary placeholder values (`A)` to `A5`) from the dropdown options.

### Button Behavior
1. Debug the **Add Bond** button functionality to identify why two clicks are required to update the 3D plot.
2. Ensure the 3D plot updates immediately after a single click by rechecking:
   - The state management logic.
   - The rendering trigger for the 3D plot in `/src/App.js` or `/src/components`.

---

### File: `/src/components/EditMolecule.jsx`
- **Dropdown Logic:**
  - Remove unnecessary placeholder values from the bond dropdown options.

### File: `/src/App.js` or `/src/components`
- **3D Plot Update:**
  - Ensure the plot updates immediately after adding a bond.

---

## Testing

- Test the **Add Bond** dropdown to confirm that only valid options are available.
- Verify that clicking the **Add Bond** button once correctly updates the 3D plot for all valid bond selections.
- Test the changes across different molecules and bond types to ensure consistency.

**Comment by shivanesh1495** [comment]
@TharunKumarrA i want to take up this issue

**Comment by maharaja1469** [comment]
Hi @TharunKumarrA was able to replicate the issue in my local machine, will debug the issue.
Meanwhile just a follow up question
1)When user didnt add any atoms, hence the dropdown initially should have no list items, can we have a placeholder label:
"from" and "to", since initial state no listitems would be populated
<img width="378" alt="image" src="https://github.com/user-attachments/assets/0a6bdb05-1916-40cb-bdb4-5f2e98ed07d0" />

**Comment by TharunKumarrA** [comment]
@maharaja1469 have you started working on the issue? because this issue is assigned to no one yet? I can assign it you if you have started working on it

**Comment by maharaja1469** [comment]
@TharunKumarrA yes looking in to the issue.

**Comment by TharunKumarrA** [comment]
Okay

**Comment by maharaja1469** [comment]
Hi @TharunKumarrA can you also pls confirm: If we where to remove unnecessary placeholders like A1 to A5 When user didnt add any atoms, hence the dropdown initially should have no list items, is this the expected behaviour

**Comment by TharunKumarrA** [comment]
Not really, have a dummy placeholders for from and to as you suggested earlier

**Comment by adithya-menon-r** [comment]
@TharunKumarrA If there is no progress, I would like to work on this issue...
